src/components/DeliverableView/InternalComment.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { Grid, Typography } from '@mui/material';
+import TextField from '@terraware/web-components/components/Textfield/Textfield';
+import strings from 'src/strings';
+import theme from 'src/theme';
+import { Deliverable } from 'src/types/Deliverables';
+import useSnackbar from 'src/utils/useSnackbar';
+import Button from 'src/components/common/button/Button';
+import DialogBox from 'src/components/common/DialogBox/DialogBox';
+import useUpdateDeliverable from 'src/scenes/AcceleratorRouter/useUpdateDeliverable';
+
+interface InternalCommentProps {
+  deliverable: Deliverable;
+}
+
+const InternalComment = ({ deliverable }: InternalCommentProps) => {
+  const snackbar = useSnackbar();
+  const { status, update } = useUpdateDeliverable();
+
+  const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false);
+  const [internalComment, setInternalComment] = useState(deliverable.internalComment || '');
+
+  const toggleDialog = useCallback(() => {
+    setIsDialogOpen(!isDialogOpen);
+  }, [isDialogOpen]);
+
+  const handleUpdate = () => {
+    update({ id: deliverable.id, internalComment });
```

Comment:
on second thought, we could have the updateDeliverable accept the entire deliverable as the payload.  
Or optionally here, destructure an UpdateRequest from `deliverable` and use that as payload, so we don't lose the `reason` value. And we can remove the status check in the service function. Assume status is always valid.
```
const { ...data }: UpdateRequest = deliverable;
update({ ...data, internalComment });
```
Thoughts on either approach?

Reply:
Yea I guess I kinda wrote this as a PATCH but we generally only use PUT. I will update to accept the entire deliverable since that is more semantic to a PUT.